zerver/lib/bugdown/__init__.py

```diff
@@ -678,32 +678,43 @@ def handleMatch(self, match):
         img.set('alt', email)
         return img
 
-emoji_tree = os.path.join(settings.STATIC_ROOT, "generated", "emoji", "images", "emoji")
-path_to_emoji = os.path.join(emoji_tree, '*.png')
-path_to_unicode_emoji = os.path.join(emoji_tree, 'unicode', '*.png')
 path_to_name_to_codepoint = os.path.join(settings.STATIC_ROOT, "generated", "emoji", "name_to_codepoint.json")
 
-unicode_emoji_list = [os.path.splitext(os.path.basename(fn))[0] for fn in glob.glob(path_to_unicode_emoji)]
 name_to_codepoint = ujson.load(open(path_to_name_to_codepoint))
+unicode_emoji_list = [name_to_codepoint[name] for name in name_to_codepoint]
 
 
-def make_emoji(emoji_name, src, display_string):
-    # type: (Text, Text, Text) -> Element
+def make_emoji(codepoint, display_string):
+    # type: (Text, Text) -> Element
+    span = markdown.util.etree.Element('span')
+    span.set('class', 'emoji emoji-%s' % (codepoint,))
```

Comment:
Can these `span`s also have an `alt`? To make it easier to implement the "don't show me any emoji" setting.

Reply:
We can just hide the spans.  The "alt text" is the actual text in the span.  So I think we're good on accessibility.